test/parallel/test-fs-writefile-with-fd.js

```diff
@@ -55,3 +55,29 @@ tmpdir.refresh();
     }));
   }));
 }
+
+// Test read-only file descriptor
+{
+  const file = join(tmpdir.path, 'test.txt');
+  fs.open(file, 'r', common.mustSucceed((fd) => {
+    fs.writeFile(fd, 'World', common.expectsError({
+      code: 'EBADF',
+      message: 'EBADF: bad file descriptor, write'
```

Comment:
@pd4d10 This test is failing for Windows. Could you please update it to expect an `EPERM` for Windows?

Reply:
OK. Updated

And added this information to https://github.com/nodejs/node/issues/38607